site/src/components/EyeButton/EyeButton.tsx

```diff
@@ -0,0 +1,72 @@
+import IconButton from "@mui/material/Button";
+import Tooltip from "@mui/material/Tooltip";
+import { css, type Interpolation, type Theme } from "@emotion/react";
+import { forwardRef, type ReactNode } from "react";
+import { EyeIcon } from "../Icons/EyeIcon";
+import { EyeClosedIcon } from "../Icons/EyeClosedIcon";
+
+interface EyeButtonProps {
+  children?: ReactNode;
+  wrapperStyles?: Interpolation<Theme>;
+  buttonStyles?: Interpolation<Theme>;
+  tooltipTitle?: string;
+  toShow:  boolean;
+  toggleHide: () => void;
```

Comment:
I'm not fully sure if a full, separate component is warranted (it might be – I'd need to check through the codebase), but at the very least, I feel like `toggleHide` is too domain-specific and not flexible enough.

Luckily, this should be easy to switch with a more general `onClick` prop